internal/provider/waypoint/resource_waypoint_add_on_definition_test.go

```diff
@@ -0,0 +1,141 @@
+					resource.TestCheckResourceAttr(resourceName, "name", updatedName),
+				),
+			},
+		},
+	})
+}
+
+// simple attribute check on the add-on definition received from the API
+func testAccCheckWaypointAddOnDefinitionName(t *testing.T, addOnDefinitionModel *waypoint.AddOnDefinitionResourceModel, nameValue string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if addOnDefinitionModel.Name.ValueString() != nameValue {
+			return fmt.Errorf("expected add-on definition name to be %s, but got %s", nameValue, addOnDefinitionModel.Name.ValueString())
+		}
+		return nil
+	}
+}
+
+func testAccCheckWaypointAddOnDefinitionExists(t *testing.T, resourceName string, definitionModel *waypoint.AddOnDefinitionResourceModel) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		// find the corresponding state object
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		client := acctest.HCPClients(t)
+		// Get the project ID and ID from state
+		projectID := rs.Primary.Attributes["project_id"]
+		appTempID := rs.Primary.Attributes["id"]
+		orgID := client.Config.OrganizationID
+
+		loc := &sharedmodels.HashicorpCloudLocationLocation{
+			OrganizationID: orgID,
+			ProjectID:      projectID,
+		}
+
+		// Fetch the add-on definition
+		definition, err := clients.GetAddOnDefinitionByID(context.Background(), client, loc, appTempID)
+		if err != nil {
+			return err
+		}
+
+		// at this time we're only verifing existence and not checking all the
+		// values, so only set name,id, and project id for now
+		definitionModel.Name = types.StringValue(definition.Name)
+		definitionModel.ID = types.StringValue(definition.ID)
+		definitionModel.ProjectID = types.StringValue(projectID)
+
+		return nil
+	}
+}
+
+func testAccCheckWaypointAddOnDefinitionDestroy(t *testing.T, definitionModel *waypoint.AddOnDefinitionResourceModel) resource.TestCheckFunc {
+	return func(_ *terraform.State) error {
+		client := acctest.HCPClients(t)
+		id := definitionModel.ID.ValueString()
+		projectID := definitionModel.ProjectID.ValueString()
+		orgID := client.Config.OrganizationID
+
+		loc := &sharedmodels.HashicorpCloudLocationLocation{
+			OrganizationID: orgID,
+			ProjectID:      projectID,
+		}
+
+		definition, err := clients.GetAddOnDefinitionByID(context.Background(), client, loc, id)
+		if err != nil {
+			// expected
+			if clients.IsResponseCodeNotFound(err) {
+				return nil
+			}
+			return err
+		}
+
+		// fall through, we expect a not found above but if we get this far then
+		// the test should fail
+		if definition != nil {
+			return fmt.Errorf("expected add-on definition to be destroyed, but it still exists")
+		}
+
+		return fmt.Errorf("both definition and error were nil in destroy check, this should not happen")
+	}
+}
+
+// TODO: (Henry) Add remaining add-on definition fields to test (tags, labels, readmemarkdown.. etc)
```

Comment:
`tags` is mentioned in this comment, but are we supporting tags yet? I didn't see that in the model struct.

This reminds me I might need to remove tags from the CLI flags for now...

Reply:
No we're not supporting `tags`, its more of a reminder to do it when we eventually add `tags` to the provider.